File: dart/lib/bebop_handlers/player_joined.dart
import 'bebop_handler.dart';
import 'package:lean_dart_client/classes/player.dart';
import 'package:lean_dart_client/schemas.g.dart';

class PlayerJoinedMsgHandler extends BebopHandler {
  static String opponentName;
  @override
  void handleMessage(sender, message) {
    message = PlayerJoinedMsg.decode(message);
    print('${message.playerName} joined');
    PlayerJoinedMsgHandler.opponentName = message.playerName;
    (sender as Player).setReady();
  }
}

File: typescript/src/bebop_handlers/playerJoined.ts
import Player from '../classes/player';
import { PlayerJoinedMsg } from '../models/pactheman.models';
import BebopHandler from './bebopHandler';

export default class PlayerJoinedMsgHandler implements BebopHandler {
    static opponentName: string;

    handleMessage(sender: any, message: any): void {
        message = PlayerJoinedMsg.decode(message);
        console.log(`${message.playerName} joined`);
        PlayerJoinedMsgHandler.opponentName = message.playerName ?? "";
        (sender as Player).setReady();
    }
}